client/src/panels/panel_declarations/finances/auth_exp_planned_spending/auth_exp_planned_spending.js

```diff
@@ -398,28 +426,37 @@ const calculate = function (subject, info, options) {
     })
     .value();
 
+  const last_shared_index = _.min([exp_values.length, auth_values.length]) - 1;
+
+  const hist_unspent_avg_pct =
+    _.reduce(
+      exp_values,
+      (result, value, i) => result + auth_values[i] - value,
+      0
+    ) /
+    _.reduce(
+      auth_values,
+      (result, value, index) =>
+        index <= last_shared_index ? result + value : result,
```

Comment:
what's the reason for checking `last_shared_index` here?

Reply:
![image](https://user-images.githubusercontent.com/25855114/88831645-a108ca80-d19d-11ea-8f62-314644316e65.png)
The text is based on the range from start to end of the blue as shown in the photo, so we don't want to use any auth values past the blue line if the calculation is based on the blue line as well.